### PR TITLE
fix(agent): add backoff in waiting pktvisor be in a running state to …

### PR DESCRIPTION
**Pull request by lpegoraro**
…send capabilities.

### CODE REVIEW

File: agent/backend/pktvisor/pktvisor.go [hunk 1]
```diff
@@ -412,11 +412,37 @@ func (p *pktvisorBackend) scrapeOpenTelemetry() (err error) {
...
 	}
 
-	err = p.exporter.Start(ctx, nil)
-	if err != nil {
-		p.logger.Error("otel mqtt exporter startup error", zap.Error(err))
-		return err
-	}
+	var errStartExp error
+	go func() {
+		startExpCtx := context.WithValue(ctx, "routine", "startExposrter")
```
Comment (lpegoraro) [inline agent/backend/pktvisor/pktvisor.go:417]: I believe you will want a cancelFunction here also, this way we will handle things better with cancelling the context after the correct initialization of the Exporters, also there is a typo in `startExposrter`

Comment (lpegoraro) [reply agent/backend/pktvisor/pktvisor.go:417]: And don' t forget to defer the call so the go func returning would cancel the context.

File: agent/backend/pktvisor/pktvisor.go [hunk 2]
```diff
@@ -412,11 +412,37 @@ func (p *pktvisorBackend) scrapeOpenTelemetry() (err error) {
...
+					p.logger.Info("waiting until mqtt client is connected", zap.String("wait time", (time.Duration(i)*time.Second).String()))
+					time.Sleep(time.Duration(i) * time.Second)
+					continue
+				}
+			}
+		}
+		if !ok {
+			p.logger.Error("mqtt did not established a connection, stopping agent")
+			p.Stop(startExpCtx)
+			startExpCtx.Done()
```
Comment (lpegoraro) [inline agent/backend/pktvisor/pktvisor.go:442]: awesome!
With the cancelFunction, we can call it in this if instead of adding the ctx.Done().

**Review by lpegoraro — COMMENTED** [review]
@mclcavalcante great stuff, just a few comments.

**Review by lpegoraro — COMMENTED** [review]
@mclcavalcante  I believe that context thing is enough so we can follow up. Great job!

**Review by manrodrigues — DISMISSED** [review]
LGTM

**Review by manrodrigues — APPROVED** [review]
LGTM

**Pull request merged**